modin/experimental/engines/omnisci_on_ray/frame/df_algebra.py

```diff
@@ -101,25 +325,94 @@ def _prints_input(self, prefix):
 
 
 class FrameNode(DFAlgNode):
-    """FrameNode holds a list of Ray object ids for frame partitions"""
+    """
+    A node to reference a materialized frame.
+
+    Parameters
+    ----------
+    modin_frame : OmnisciOnRayFrame
+        Referenced frame.
+
+    Attributes
+    ----------
+    modin_frame : OmnisciOnRayFrame
+        Referenced frame.
+    """
 
     def __init__(self, modin_frame):
         self.modin_frame = modin_frame
 
     def copy(self):
+        """
+        Make a shallow copy of the node.
+
+        Returns
+        -------
+        FrameNode
+        """
         return FrameNode(self.modin_frame)
 
     def _append_partitions(self, partitions):
+        """
+        Append all partitions of the referenced frame to `partitions` list.
+
+        Parameters
+        ----------
+        partitions : list
+            Output list of partitions.
+        """
         partitions += self.modin_frame._partitions.flatten()
 
     def _append_frames(self, frames):
+        """
+        Append the referenced frame to `frames` list.
+
+        Parameters
+        ----------
+        frames : list
+            Output list of frames.
+        """
         frames.append(self.modin_frame)
 
     def _prints(self, prefix):
+        """
+        Return a string representation of the tree.
+
+        Parameters
+        ----------
+        prefix : str
+            A prefix to add at each string of the dump.
+
+        Returns
+        -------
+        str
+        """
         return f"{prefix}{self.modin_frame.id_str()}"
 
 
 class MaskNode(DFAlgNode):
+    """
+    A filtering node which filters rows by index values or row id.
+
+    Parameters
+    ----------
+    base : DFAlgNode
+        A filtered frame.
+    row_indices : list or None
```

Comment:
It should be noticed here, that `row_indices ` and `row_numeric_idx ` are optional.

Reply:
Is it the same actually? I'd interpret 'optional' as 'object can miss this attribute'. It's not the same as having an attribute set to `None`.

Reply:
Fixed